specifyweb/frontend/js_src/lib/components/WorkBench/index.tsx

```diff
@@ -0,0 +1,94 @@
+import React from 'react';
+import { useNavigate, useParams } from 'react-router-dom';
+
+import { useAsyncState } from '../../hooks/useAsyncState';
+import { useBooleanState } from '../../hooks/useBooleanState';
+import { useErrorContext } from '../../hooks/useErrorContext';
+import { commonText } from '../../localization/common';
+import { wbText } from '../../localization/workbench';
+import { ajax } from '../../utils/ajax';
+import { f } from '../../utils/functools';
+import type { GetSet } from '../../utils/types';
+import { className } from '../Atoms/className';
+import { LoadingContext } from '../Core/Contexts';
+import { LoadingScreen } from '../Molecules/Dialog';
+import { useMenuItem } from '../Header/MenuContext';
+import { treeRanksPromise } from '../InitialContext/treeRanks';
+import { Dialog } from '../Molecules/Dialog';
+import { NotFoundView } from '../Router/NotFoundView';
+import type { Dataset } from '../WbPlanView/Wrapped';
+import { WbView } from './WbView';
+
+const fetchTreeRanks = async (): Promise<true> => treeRanksPromise.then(f.true);
+
+const fetchDataSet = async (
+  datasetId: number | undefined
+): Promise<Dataset | undefined> =>
+  typeof datasetId === 'number'
```

Comment:
nitpick: could you move the `typeof datasetId === 'number' check into the useDataSet hook?

reasons:
- this function is for fetching dataset. if you don't have the data you need to fetch a dataset, you should not be calling the function. on the other hand, if I see a function fetchDataSet in the code, I would expect it to always return a dataset, unless some error has occurred
- not the case here, but imagine that I am trying to use fetchDataSet in some place where datasetId is definitely defined - even though datasetId is defined, the return type of `fetchDataSet` still says that it could return undefined, which in turn makes typescript force me to handle the case when `fetchDataSet` returned undefined even though I know that should never happen - bad developer experience

so generally, undefined checks should be done before calling a function. exception: if the undefined check is complicated, or if the function is going to be called with a maybe-undefined often, making it more convenient to do the undefined check inside the function rather than outside